src/core/u-compress.c

```diff
@@ -355,9 +406,10 @@ REBSER *Decompress(
 
     // Since the initialization succeeded, go ahead and make the output buffer
     //
-    REBSER *output = Make_Binary(buf_size);
+    REBSER *output = Make_Binary(sizeof(REBSER*) + buf_size);
+    *cast(REBSER**, BIN_HEAD(output)) = output; // self-referential prefix
     strm.avail_out = buf_size;
-    strm.next_out = BIN_HEAD(output);
+    strm.next_out = BIN_HEAD(output) + sizeof(REBSER*);
 
```

Comment:
Maybe...glad you're interested enough to think about it.  Someday, all of this will be yours, son. 
 ;-P

Though given how many new features got grafted into the series code, I'd really like there to be a big "going-over" of all the basic series code to streamline what's there now.  It's a bit of a testament to how good the asserts and checks are that the whole thing runs as well as it does! 
 But things like SERIES_INFO_HAS_DYNAMIC were organic outgrowths on some relatively twisty original code, and there's tons of simplifications that could be done...there's likely a lot of inefficiency in there at the moment.  (cpu-cycle-wise, though space-wise it's pretty darn efficient)

For now, it's probably better to just keep the mechanic isolated to those 4 routines, use them in and Rebol_Realloc() directly in the compression code shared between LodePNG and COMPRESS/DECOMPRESS, and not reach beneath the REBSER level.  We don't have to worry about anyone copying the series because the abstraction layer doesn't expose the REBSER until you're supposed to have all the pointer-based clients gone *(Rebserize should in fact corrupt the REBSER* before biasing it out, doing that now...)*